submissions/antonkottans/frogger/app.js

```diff
@@ -1,71 +1,93 @@
-const Enemy = function (x, y, speed) {
-    this.x = x;
-    this.y = y;
-    this.speed = speed;
-    this.sprite = "images/enemy-bug.png";
-};
-const field = {
-    x: 4,
-    y: 4,
-};
+class Field {
+    constructor() {
+        this.maxreachableXCoord = 4;
+        this.maxreachableYCoord = 4;
+        this.topIndent = 50;
+        this.SquareWidth = 100;
+        this.SquareHeight = 80;
+    }
+}
+const field = new Field();
+
+class Enemy {
+    constructor({ x, y, speed, row }) {
+        this.x = x;
+        this.y = y;
+        this.speed = speed;
+        this.sprite = "images/enemy-bug.png";
+        this.row = row;
+    }
+    checkCollision() {
+        const distanceOfCollision = 70;
```

Comment:
How you get this value?)

Reply:
its difference in x-coord  of bug and player while visual connection

Reply:
I think this approach is not flexible, it is better to check collisions with respect to coordinates not only on the x-axis and on the y-axis as well.